cmd/gokr-packer/packer.go

```diff
@@ -281,6 +303,130 @@ func derivePartUUID(hostname string) uint32 {
 	return h.Sum32()
 }
 
+func generateAndSignCert() ([]byte, *rsa.PrivateKey, error) {
+	notBefore := time.Now()
+	// TODO: Make duration configurable?
+	notAfter := notBefore.Add(2 * 365 * 24 * time.Hour)
+	serialNumberLimit := new(big.Int).Lsh(big.NewInt(1), 128)
+	serialNumber, err := rand.Int(rand.Reader, serialNumberLimit)
+	if err != nil {
+		return nil, nil, err
+	}
+	template := x509.Certificate{
+		SerialNumber: serialNumber,
+		Subject: pkix.Name{
+			//TODO: Configurable?
+			Organization: []string{"gokrazy"},
+		},
+		NotBefore:             notBefore,
+		NotAfter:              notAfter,
+		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		BasicConstraintsValid: true,
+		DNSNames:              []string{*hostname},
+	}
+	priv, err := rsa.GenerateKey(rand.Reader, 4096)
+	if err != nil {
+		return nil, nil, err
+	}
+	pub := &priv.PublicKey
+	derBytes, err := x509.CreateCertificate(rand.Reader, &template, &template, pub, priv)
+	if err != nil {
+		return nil, nil, err
+	}
+	return derBytes, priv, err
+}
+
+func getCertificate() (string, string, error) {
+	hostConfigPath := config.HostnameSpecific(*hostname)
+	var certPath, keyPath string
+	switch *useTLS {
+	case "self-signed":
+		certPath = path.Join(string(hostConfigPath), "cert.pem")
```

Comment:
This code branch is pretty long. Can you factor it into its own function please? That should allow you to use early returns to make this a little cleaner.